ros/lib/utils.py

```diff
@@ -151,15 +151,37 @@ def insert_performance_profiles(session, system_id, fields):
        performance_profile_history table.
     """
     fields = {} if fields is None else fields
-    session.execute(db.delete(PerformanceProfile).filter(PerformanceProfile.system_id == system_id))
-    session.commit()
+    fields_for_perf_profile_history = redact_instance_type_and_price(fields)
 
-    for model_class in [PerformanceProfile, PerformanceProfileHistory]:
-        new_entry = model_class(**fields)
+    old_profile_record = session.query(PerformanceProfile).filter_by(
+        system_id=system_id).first()
+    if old_profile_record:
+        session.delete(old_profile_record)
+        session.commit()
+
+    model_fields_map = {PerformanceProfile: fields,
+                        PerformanceProfileHistory: fields_for_perf_profile_history}
+
+    for model_class, model_fields in model_fields_map.items():
+        new_entry = model_class(**model_fields)
         session.add(new_entry)
         session.flush()
 
 
+def redact_instance_type_and_price(fields):
```

Comment:
(Based on my above comment) If we add those fields to the history table then we don't require this function and any changes to `insert_performance_profiles`.